src/settings-hardening.php

```diff
@@ -552,6 +552,58 @@ public static function fileeditor()
         return self::drawSection($params);
     }
 
+    /**
+     * Checks if the Automatic Secret Keys Updater is enabled.
+     *
+     * Changing the Secret Keys will invalidate all existing cookies, forcing all
+     * logged in users to login again. Doing this frequently will decrease the
+     * chances of misuse of sessions left open on unprotected devices.
+     *
+     * @return string HTML code with the replaced template variables
+     */
+    public static function autoSecretKeyUpdater()
+    {
+        $cronName = 'sucuriscan_autoseckeyupdater';
+        $cronFrequency = 'weekly';
+
+        // Activate auto secret key update.
+        if (self::processRequest(__FUNCTION__)) {
+            if (SucuriScanEvent::addScheduledTask($cronName, $cronFrequency)) {
+                SucuriScanInterface::info(__('Automatic Secret Keys Updater enabled.', 'sucuri-scanner'));
+                SucuriScanEvent::reportNoticeEvent(__('Automatic Secret Keys Updater enabled.', 'sucuri-scanner'));
+            } else {
+                SucuriScanInterface::error(__('Something went wrong.', 'sucuri-scanner'));
+            }
+        }
+
+        // Deactivate auto secret key update.
+        if (self::processRequest(__FUNCTION__ . '_revert')) {
+            if (SucuriScanEvent::deleteScheduledTask($cronName)) {
+                SucuriScanInterface::info(__('Automatic Secret Keys Updater disabled.', 'sucuri-scanner'));
+                SucuriScanEvent::reportNoticeEvent(__('Automatic Secret Keys Updater disabled.', 'sucuri-scanner'));
+            } else {
+                SucuriScanInterface::error(__('Something went wrong.', 'sucuri-scanner'));
+            }
+        }
+        
+        // Set status of auto secret key update.
+        $params = array();
+        $params['Hardening.Title'] = __('Automatic Secret Keys Update', 'sucuri-scanner');
+        $params['Hardening.Description'] = __('Changing the Secret Keys will invalidate all existing cookies, forcing all logged in users to login again. Doing this frequently will decrease the chances of misuse of sessions left open on unprotected devices.', 'sucuri-scanner');
+
+        $params['Hardening.Status'] = 0;
+        $params['Hardening.FieldName'] = __FUNCTION__;
+        $params['Hardening.FieldText'] = __('Apply Hardening', 'sucuri-scanner');
```

Comment:
Applying this hardening will schedule the key update for every week, but it doesn't say anywhere on this page how often it's going to be.